enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -115,211 +128,48 @@ service will include a database; sqlite if possible, or else postgresql.
+In the meantime, status will be propagated to the appropriate k8s resource when
+an agent reports relevant information.
 
 **AgentDrivenInstallationImage**
 This resource represents a discovery image that should be used for booting
 hosts. In the REST API this resource is embedded in the "cluster" resource, but
 for kubernetes-native is more natural for it to be separate.
 
-Spec:
- * clusterName: string
- * sshPublicKey: string
- * pullSecretName: string (reference to k8s secret)
- * ignitionOverrides: object
- * staticIpConfiguration: object
-
-Status:
- * state: string
- * sizeBytes: integer
- * downloadUrl: string
- * createdAt: string
- * expiresAt: string
-
-**AgentDrivenClusterInstallation**
-This resource correlates to the "cluster" resource in the assisted installer's
-current API.
-
-Spec:
- * name: string
- * approved: boolean (used for starting installation)
- * openshiftVersion: string
- * baseDnsDomain: string
- * clusterNetworkCidr: string
- * clusterNetworkHostPrefix: integer
- * serviceNetworkCidr: string
- * apiVip: string
- * apiVipDnsName: string
- * ingressVip: string
- * machineNetworkCidr: string
- * sshPublicKey: string
- * vipDhcpAllocation: boolean
- * httpProxy: string
- * httpsProxy: string
- * noProxy: string
- * userManagedNetworking: boolean
- * additionalNtpSource: string
- * installConfigOverrides: string
-
-Status:
-* state: string enum
-* stateInfo: string
-* host_networks: array of (cidr, host_ids)
-* install_started_at: string
-* hosts: int
-* progress:
-  * progress_info: string
-  * progress_updated_at: string
-* validation_info:
-  * configuration:
-    * is_pull_secret_set: [success, failure, pending, error]
-  * host_data:
-    * all_hosts_are_ready_to_install: [success, failure, pending, error]
-    * sufficient_masters_count: [success, failure, pending, error]
-  * network:
-    * machine_cidr_defined: [success, failure, pending, error]
-    * cluster_cidr_defined: [success, failure, pending, error]
-    * service_cidr_defined: [success, failure, pending, error]
-    * no_cidrs_overlapping: [success, failure, pending, error]
-    * network_prefix_valid: [success, failure, pending, error]
-    * machine_cidr_equals_to_calculated_cidr: [success, failure, pending, error]
-    * api_vip_defined: [success, failure, pending, error]
-    * api_vip_valid: [success, failure, pending, error]
-    * ingress_vip_defined: [success, failure, pending, error]
-    * ingress_vip_valid: [success, failure, pending, error]
-    * dns_domain_defined: [success, failure, pending, error]
-    * ntp_server_configured: [success, failure, pending, error]
-* connectivity_majority_groups: string
-* updated_at: string
-* created_at: string
-* install_started_at: string
-* install_completed_at: string
-* controller_logs_collected_at: string
+A first draft of this CRD has been
+[implemented](https://github.com/openshift/assisted-service/blob/master/internal/controller/api/v1alpha1/image_types.go)
```

Comment:
It's a bit odd to have code approved before the design.

Reply:
As a matter of practicality, reviewing the field-level details of the API seems to be most productive when looking at the go structs. And since the approach is currently being proved out through code, it doesn't seem productive to try duplicating those details here and keeping them in sync. Open to suggestions though.

Reply:
I have noticed a tendency of the edge teams to rush to implementation before the broader organization is aligned with the direction, resulting in re-implementing some changes as many as 3 times (see #555 for example). We may all work more efficiently if we focus on agreement here first.